docs/reqcontext.rst

```diff
@@ -175,6 +200,45 @@ Certain parts of the test system might also temporarily create a request
 context without calling the before-request handlers.  Make sure to write
 your teardown-request handlers in a way that they will never fail.
 
+.. _ensuring-teardown-callbacks-succeed:
+
+Ensuring teardown callbacks succeed
+-----------------------------------
+
+.. admonition:: Watch Out
+
+    Teardown callbacks must never raise. If they do, teardown callbacks that
+    haven't run yet won't run, and request/app contexts won't pop, which will
+    cause memory/file leaks at best and context persisting into later requests
+    at worst.
+
+    Note that extensions can register teardown callbacks, and those callbacks
```

Comment:
Not sure how much I like this section.. as a new user I'd probably think one of those two things when reading it:

> *This framework sucks, they can't properly handle errors so I need to almost-monkeypatch it*

> *Are Flask extensions really so bad that they don't handle exceptions like they should?*